src/qibolab/platforms/abstract.py

```diff
@@ -195,53 +176,120 @@ def update(self, updates: dict):
+                    self.settings["native_gates"]["single_qubit"][qubit]["RX"]["duration"] = duration
+
+                # ramsey / spin_echo
+                elif par == "t2" or par == "t2_spin_echo":
+                    t2 = float(value)
+                    self.qubits[qubit].T2 = t2
+                    self.settings["characterization"]["single_qubit"][qubit]["T2"] = t2
+
+                # t1
+                elif par == "t1":
+                    t1 = float(value)
+                    self.qubits[qubit].T1 = t1
+                    self.settings["characterization"]["single_qubit"][qubit]["T1"] = t1
+
+                # classification
+                elif par == "threshold":
+                    threshold = float(value)
+                    self.qubits[qubit].thresold = threshold
+                    self.settings["characterization"]["single_qubit"][qubit]["threshold"] = threshold
+
+                # classification
+                elif par == "iq_angle":
+                    iq_angle = float(value)
+                    self.qubits[qubit].iq_angle = iq_angle
+                    self.settings["characterization"]["single_qubit"][qubit]["iq_angle"] = iq_angle
+
+                # classification
+                elif par == "mean_gnd_states":
+                    mean_gnd_states = str(value)
+                    self.qubits[qubit].iq_angle = iq_angle
+                    self.settings["characterization"]["single_qubit"][qubit]["mean_gnd_states"] = mean_gnd_states
+
+                # classification
+                elif par == "mean_exc_states":
+                    mean_exc_states = str(value)
+                    self.qubits[qubit].iq_angle = iq_angle
+                    self.settings["characterization"]["single_qubit"][qubit]["mean_exc_states"] = mean_exc_states
+
+                # drag pulse tunning
                 elif "beta" in par:
                     shape = self.single_qubit_natives[qubit]["RX"]["shape"]
                     rel_sigma = re.findall(r"[\d]+[.\d]+|[\d]*[.][\d]+|[\d]+", shape)[0]
                     self.single_qubit_natives[qubit]["RX"]["shape"] = f"Drag({rel_sigma}, {float(value)})"
+                    self.settings["native_gates"]["single_qubit"][qubit]["RX"][
+                        "shape"
+                    ] = f"Drag({rel_sigma}, {float(value)})"
 
-                elif "length" in par:  # assume only drive length
-                    self.single_qubit_natives[qubit]["RX"]["duration"] = int(value)
-                elif par == "t2_spin_echo":
-                    self.qubits[qubit].T2_spin_echo = int(value)
-                elif par == "readout_attenuation":
-                    True
                 else:
-                    raise_error(ValueError, f"Unknown parameter {par}.")
+                    raise_error(ValueError, f"Unknown parameter {par} for qubit {qubit}")
+
+        # reload_settings after execute any calibration routine keeping fitted parameters
+        self.reload_settings()
```

Comment:
I prefer to keep it just in case. Maybe is there is a not detected error in a platform (I have not tested the sweepers in all platforms) and this is a double check. But if you want, we can remove it. From the point of view of performance it does not increases the execution time significantly.
If we removed it, we should test all the platforms sweepers in depth.